src/main/java/bio/terra/app/model/GoogleRegion.java

```diff
@@ -82,8 +88,10 @@ public static GoogleRegion fromValueWithDefault(String text) {
     }
 
     public static boolean matchingRegionWithFallbacks(List<StorageResource> storage, GoogleRegion region) {
-        for (StorageResource resource : storage) {
-            switch (resource.getCloudResource()) {
+        List<StorageResource> gcpStorage =
+            storage.stream().filter(s -> s.getCloudPlatform() == CloudPlatform.GCP).collect(Collectors.toList());
+        for (StorageResource resource : gcpStorage) {
+            switch ((GoogleCloudResource) resource.getCloudResource()) {
```

Comment:
Also it seems odd to collect and then iterate over the collection. Moving the check inside the `for` loop results in less code too.
```suggestion
        for (var resource : storage) {
            if (resource.getCloudPlatform() != CloudPlatform.GCP) {
                continue:
            }
            switch ((GoogleCloudResource) resource.getCloudResource()) {
```

Reply:
Yeah, that's a really good point...I'll rework this bit